usbc: Use a DT alias to identify USB-CPD port

andysan opened the pull request:
Samples sometimes need to identify a default USB-C port. Instead of hard-coding a node label, use a device tree alias  to point to the correct node.

The intention is that this will make it easier to include board-specific USB-C configuration (tcpc, vbus) in the main board configuration.

andysan committed:
usbc: Use a DT alias to identify USB-C port

Samples sometimes need to identify a default USB-C port. Instead of
hard-coding a node label, use the usbc-port0 alias to point to the
correct node.

Signed-off-by: Andreas Sandberg <andreas@sandberg.uk>

mbolivar-ampere reviewed (changes requested):
We use aliases for conventions that multiple samples want to align on. Chosen nodes are basically for when something in subsys/ needs a hint about which peripheral among multiple options it should use.

andysan commented:
> We use aliases for conventions that multiple samples want to align on. Chosen nodes are basically for when something in subsys/ needs a hint about which peripheral among multiple options it should use.

Done.

keith-zephyr commented on samples/subsys/usb_c/sink/src/main.c:
```diff
@@ -13,8 +13,8 @@
 
-#define PORT1_NODE DT_NODELABEL(port1)
-#define PORT1_POWER_ROLE		DT_ENUM_IDX(DT_NODELABEL(port1), power_role)
+#define PORT1_NODE		DT_ALIAS(usbc_port0)
```
Update the macro naming to match the alias nodename.  You could also drop the "0" suffix, but I'm fine with or without the suffix.
```suggestion
#define USBC_PORT0_NODE		DT_ALIAS(usbc_port0)
```

andysan replied:
I think I prefer to keep the 0 in the alias at least. There are boards out there with multiple USB-C ports, so I think it is conceivable that we will have examples using two ports.

keith-zephyr commented on samples/subsys/usb_c/sink/src/main.c:
```diff
@@ -13,8 +13,8 @@
-#define PORT1_NODE DT_NODELABEL(port1)
-#define PORT1_POWER_ROLE		DT_ENUM_IDX(DT_NODELABEL(port1), power_role)
+#define PORT1_NODE		DT_ALIAS(usbc_port0)
+#define PORT1_POWER_ROLE	DT_ENUM_IDX(PORT1_NODE, power_role)
```
```suggestion
#define USC_PORT0_POWER_ROLE	DT_ENUM_IDX(PORT1_NODE, power_role)
```

keith-zephyr commented on samples/subsys/usb_c/source/src/main.c:
```diff
@@ -15,8 +15,8 @@ LOG_MODULE_REGISTER(main, LOG_LEVEL_DBG);
 
-#define PORT1_NODE		DT_NODELABEL(port1)
-#define PORT1_POWER_ROLE	DT_ENUM_IDX(DT_NODELABEL(port1), power_role)
+#define PORT1_NODE		DT_ALIAS(usbc_port0)
```
Update this with the same naming convention that you decide on for the other sample.  `USBC_PORT0_NODE` or `USBC_PORT_NODE`.

andysan committed:
samples: usb_c: Rename port1 to port0 for consistency

Aliases, including usbc_port0, start counting from zero. This meant
that the DT alias is inconsistent with the port numbering in the
samples. Change the numbering scheme to start from zero to make this
consistent.

Signed-off-by: Andreas Sandberg <andreas@sandberg.uk>

andysan commented:
Ping @mbolivar-ampere
